dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -342,25 +367,33 @@ private IHTMLPage getPageByUri(final PageMode mode, final Host host, final Strin
         return htmlPage;
     }
 
+    private IHTMLPage getPageById(final PageMode mode, final Host host, final String id)
+            throws DotDataException, DotSecurityException {
+
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
```

Comment:
what happen if the request is null?

Reply:
done: https://github.com/dotCMS/core/pull/17859/commits/d8d0ac8874bfdf61503d97d02689d958d2fa12d4#diff-051d49b06feb0c7462b349b6b61988a8R378